test/gtest/pdist.hpp

```diff
@@ -0,0 +1,240 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/miopen.h>
+#include <miopen/pdist.hpp>
+#include <gtest/gtest.h>
+
+#include "get_handle.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include "cpu_pdist.hpp"
+
+struct PdistTestCase
+{
+    std::vector<size_t> dims;
+    double p;
+    bool is_contiguous;
+
+    friend std::ostream& operator<<(std::ostream& os, const PdistTestCase& tc)
+    {
+        os << "dims: ( ";
+        for(auto d : tc.dims)
+        {
+            os << d << " ";
+        }
+        os << ")";
+        os << " is_contiguous: " << tc.is_contiguous;
+        os << " p: " << tc.p;
+
+        return os;
+    }
+
+    std::vector<size_t> GetDims() const { return dims; }
+    double GetP() const { return p; }
+
+    PdistTestCase() {}
+
+    PdistTestCase(std::vector<size_t> dims_, double p_, bool is_contiguous_)
+        : dims(dims_), p(p_), is_contiguous(is_contiguous_)
+    {
+    }
+
+    std::vector<size_t> ComputeStrides() const
+    {
+        std::vector<size_t> inputDim = dims;
+        if(!is_contiguous)
+            std::swap(inputDim.front(), inputDim.back());
+        std::vector<size_t> strides(inputDim.size());
+        strides.back() = 1;
+        for(int i = inputDim.size() - 2; i >= 0; --i)
+            strides[i] = strides[i + 1] * inputDim[i + 1];
+        if(!is_contiguous)
+            std::swap(strides.front(), strides.back());
+        return strides;
+    }
+};
+
+inline std::vector<PdistTestCase> PdistTestConfigs()
+{
+    return {
+
+        PdistTestCase({2, 5}, 2.0, true),
+        PdistTestCase({2, 10}, 5.0, true),
+        PdistTestCase({5, 1}, 0.0, true),
+        PdistTestCase({5, 5}, 0.0, true),
+        PdistTestCase({10, 1}, 1.0, true),
+        PdistTestCase({10, 10}, 0.0, true),
+        PdistTestCase({100, 100}, 0.0, true),
+        PdistTestCase({100, 100}, 1.0, true),
+        PdistTestCase({100, 100}, 2.0, true),
+        PdistTestCase({100, 100}, 5.0, true),
+
+        PdistTestCase({2, 5}, HUGE_VAL, true),
+        PdistTestCase({2, 10}, HUGE_VAL, true),
+        PdistTestCase({5, 1}, HUGE_VAL, true),
+        PdistTestCase({5, 5}, HUGE_VAL, true),
+        PdistTestCase({10, 1}, HUGE_VAL, true),
+        PdistTestCase({10, 10}, HUGE_VAL, true),
+        PdistTestCase({100, 100}, HUGE_VAL, true),
+    };
+}
+
+// Remove big tests with reduction from FP16 test because the result will be overflow/ underflow
```

Comment:
This may because of the too big value from gen_descreet_uniform_sign. try 
```
auto gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 1); };
        auto output_gen_value = [](auto...) {
            return prng::gen_descreet_uniform_sign<T>(1e-2, 1);
        };
```
Then remove PdistFp16TestConfigs because of the duplicated

Reply:
Solved: I got it. I fixed as suggested and all test cases passed. I'll remove `PdistFp16TestConfigs`.